contrib/coredns/policy/policy.go

```diff
@@ -150,7 +218,13 @@ func (p *PolicyMiddleware) ServeDNS(ctx context.Context, w dns.ResponseWriter, r
 	}
 
 	if response.Permit {
-		return middleware.NextOrFailure(p.Name(), p.Next, ctx, w, r)
+		for _, rattrs := range response.Obligation {
+			if rattrs.Id == PolicyIdFieldName {
```

Comment:
If I am right about the above comment, then this just becomes response.PolicyId.